samples/nrf9160/nrf_cloud_rest_agps/src/main.c

```diff
@@ -0,0 +1,735 @@
+			BUTTON_EVENT_BTN_NUM);
+		(void)k_sem_take(&button_press_sem, K_FOREVER);
+	} else if (ret == 1) {
+		LOG_INF("Device already provisioned");
+	} else {
+		LOG_ERR("Device provisioning failed");
+	}
+	return ret;
+}
+
+static int get_mobile_network_info(int *mcc_num, int *mnc_num)
+{
+	__ASSERT_NO_MSG((mcc_num != NULL) && (mnc_num != NULL));
+	char buf[100];
+	char mcc[4];
+	char mnc[4];
+	int len;
+
+	len = modem_info_string_get(MODEM_INFO_OPERATOR, buf, sizeof(buf));
+	if (len <= 0) {
+		return len; /* it's an error code, not a length */
+	}
+	if ((len < 5) || (len > 6)) {
+		LOG_ERR("unexpected mcc+mnc string length:%d, %s", len, log_strdup(buf));
+		return -EINVAL;
+	}
+
+	/* mobile country code is required to be 3 decimal digits */
+	memcpy(mcc, buf, 3);
+	mcc[3] = '\0';
+	len -= 3;
+
+	/* mobile network code is either 2 or 3 decimal digits */
+	memcpy(mnc, &buf[3], len);
+	mnc[len] = '\0';
+
+	*mcc_num = (int)strtol(mcc, NULL, 10);
+	*mnc_num = (int)strtol(mcc, NULL, 10);
+
+	LOG_DBG("operator:%s, mcc_num:%d, mnc_num:%d", log_strdup(buf), *mcc_num, *mnc_num);
+	return 0;
+}
+
+static int configure_lna(void)
+{
+	int err;
+	const char *xmagpio_command = CONFIG_SAMPLE_AT_MAGPIO;
+	const char *xcoex0_command = CONFIG_SAMPLE_AT_COEX0;
+
+	/* Make sure the AT command is not empty */
+	if (xmagpio_command[0] != '\0') {
+		err = at_cmd_write(xmagpio_command, NULL, 0, NULL);
+		if (err) {
+			LOG_ERR("Failed to send XMAGPIO command");
+			return err;
+		}
+	}
+
+	if (xcoex0_command[0] != '\0') {
+		err = at_cmd_write(xcoex0_command, NULL, 0, NULL);
+		if (err) {
+			LOG_ERR("Failed to send XCOEX0 command");
+			return err;
+		}
+	}
+
+	return 0;
+}
+
+static int init_gnss(void)
+{
+	int err;
+	size_t buf_size = nrf_cloud_agps_get_max_file_size(false);
+
+	agps_result.buf = k_calloc(buf_size, 1);
+	if (!agps_result.buf) {
+		LOG_ERR("Failed to allocate %u bytes for A-GPS buffer", agps_result.buf_sz);
+		failure_count = FAILURE_LIMIT;
+		return -ENOMEM;
+	}
+	agps_result.buf_sz = (uint32_t)buf_size;
+
+	k_work_queue_start(
+		&agps_work_q,
+		agps_workq_stack_area,
+		K_THREAD_STACK_SIZEOF(agps_workq_stack_area),
+		AGPS_WORKQ_THREAD_PRIORITY,
+		NULL);
+
+	k_work_init(&get_agps_data_work, get_agps_data);
+
+	/* Initialize and configure GNSS */
+	err = nrf_modem_gnss_init();
+	if (err) {
+		LOG_ERR("Failed to initialize GNSS interface: %d", err);
+		return err;
+	}
+
+	err = nrf_modem_gnss_event_handler_set(gnss_event_handler);
+	if (err) {
+		LOG_ERR("Failed to set GNSS event handler: %d", err);
+		return err;
+	}
+
+	if (IS_ENABLED(CONFIG_REST_NMEA)) {
+		err = nrf_modem_gnss_nmea_mask_set(NRF_MODEM_GNSS_NMEA_RMC_MASK |
+						  NRF_MODEM_GNSS_NMEA_GGA_MASK |
+						  NRF_MODEM_GNSS_NMEA_GLL_MASK |
+						  NRF_MODEM_GNSS_NMEA_GSA_MASK |
+						  NRF_MODEM_GNSS_NMEA_GSV_MASK);
+		if (err) {
+			LOG_ERR("Failed to set GNSS NMEA mask: %d", err);
+			return err;
+		}
+	}
+
+	err = nrf_modem_gnss_fix_retry_set(120);
```

Comment:
would it be useful for this and the value used for `nrf_modem_gnss_fix_interval_set` to be config options?

Reply:
Probably.  The other samples just hard code them.